docs: Add resume recorded work direcgtory section (#1729)
seqeralabs/docs: platform-cloud/docs/launch/cache-resume.mdx

For a detailed explanation of the Nextflow resume feature, see _Demystifying Nextflow resume_ ([Part 1](https://www.nextflow.io/blog/2019/demystifying-nextflow-resume.html) and [Part 2](https://www.nextflow.io/blog/2019/troubleshooting-nextflow-resume.html)) in the Nextflow blog.
:::

### Change compute environment during run resume

If you have the required permissions, you can change the compute environment when you resume a run. The new compute environment must have access to the work directory that the original run used.

Platform records the work directory when you submit a run, then compares that recorded value against the work directory configured on each compute environment. A compute environment is eligible only if its work directory is the same as the recorded value or a parent of that value.

Only the run's recorded value is frozen. Platform reads each compute environment's work directory from its current configuration. Changing the work directory in the pipeline's configuration after you submit a run does not affect which compute environments that run can resume on.

> Ineligible compute environments appear in the **Compute environment** drop-down but you cannot select them. The drop-down does not state why. If the original compute environment no longer exists, Platform also reports that compute environment as unavailable. That notice concerns only the missing compute environment ID. It does not indicate a work directory mismatch.

Reviewer: ```suggestion
If the original compute environment no longer exists, Platform displays it as unavailable. The unavailability only refers to the missing compute environment ID. It does not indicate a work directory mismatch.
```

> Ineligible compute environments appear in the **Compute environment** drop-down but you cannot select them. The drop-down does not state why. If the original compute environment no longer exists, Platform also reports that compute environment as unavailable. That notice concerns only the missing compute environment ID. It does not indicate a work directory mismatch.

Reviewer 2: Don't need the 'The drop down does not state why' here IMO
